src/software/ai/navigator/path_planner/theta_star_path_planner.cpp

```diff
@@ -34,133 +33,132 @@ bool ThetaStarPathPlanner::isUnBlocked(int row, int col)
-                       calculateHValue(parent_i, parent_j, pNew);
+                parent_new = parent;
+                g_new      = cell_heuristics[parent.row()][parent.col()].g +
+                        calculateHValue(parent, new_coord);
             }
             else
             {
-                new_parent_i = pCurr.first;
-                new_parent_j = pCurr.second;
-                gNew = cellDetails[pCurr.first][pCurr.second].g_ + currToNextNodeDist;
+                parent_new = current_coord;
+                g_new      = cell_heuristics[current_coord.row()][current_coord.col()].g +
+                        curr_to_new_dist;
             }
 
-            double hNew = calculateHValue(pNew.first, pNew.second, dest);
-            double fNew = gNew + hNew;
+            double h_new = calculateHValue(new_coord, dest);
+            double f_new = g_new + h_new;
 
             // If it isn’t on the open list, add it to
             // the open list. Make the current square
             // the parent of this square. Record the
-            // f, g, and h costs of the square GridCell
+            // f, g, and h costs of the square CellHeuristic
             //			 OR
             // If it is on the open list already, check
             // to see if this path to that square is better,
             // using 'f' cost as the measure.
-            if (cellDetails[pNew.first][pNew.second].f_ == DBL_MAX ||
-                cellDetails[pNew.first][pNew.second].f_ > fNew)
+            if (cell_heuristics[new_coord.row()][new_coord.col()].f == DBL_MAX ||
+                cell_heuristics[new_coord.row()][new_coord.col()].f > f_new)
             {
-                openList.insert(
-                    std::make_pair(fNew, std::make_pair(pNew.first, pNew.second)));
-
-                // Update the details of this GridCell
-                cellDetails[pNew.first][pNew.second].f_        = fNew;
-                cellDetails[pNew.first][pNew.second].g_        = gNew;
-                cellDetails[pNew.first][pNew.second].h_        = hNew;
-                cellDetails[pNew.first][pNew.second].parent_i_ = new_parent_i;
-                cellDetails[pNew.first][pNew.second].parent_j_ = new_parent_j;
+                open_list.insert(new_coord);
+
+                // Update the details of this CellHeuristic
+                cell_heuristics[new_coord.row()][new_coord.col()].f      = f_new;
+                cell_heuristics[new_coord.row()][new_coord.col()].g      = g_new;
+                cell_heuristics[new_coord.row()][new_coord.col()].h      = h_new;
+                cell_heuristics[new_coord.row()][new_coord.col()].parent = parent_new;
             }
-            // If the destination GridCell is the same as the
+            // If the destination CellHeuristic is the same as the
             // current successor
-            if (isDestination(pNew.first, pNew.second, dest) == true)
+            if (isDestination(new_coord, dest) == true)
```

Comment:
Were we going to replace `isDestination` with `==`?

Reply:
Yeah we can do this because Coordinates are pairs so == is well defined

Reply:
We're not using `==` here for the coordinates?